### PR TITLE
Remove sensitive identifiers from Github Actions and Terraform

### LINKED ISSUE #10: Move sensitive identifiers from Github Actions workflow into secrets.
AWS account IDs are considered [sensitive information](https://github.com/cisagov/development-guide/blob/develop/open-source-policy/practice.md#protecting-sensitive-information).  

https://github.com/cisagov/domain-manager-cicd/blob/aa48571a8a62570e019294663241f174d2a160f7/.github/workflows/cool.yml#L11-L12

Please remove these from the repository and store them as [encrypted secrets](https://docs.github.com/en/actions/reference/encrypted-secrets#about-encrypted-secrets).

### PR DESCRIPTION
**Pull request by zenine07**
#  #

## 🗣 Description ##

* Move accounts ids to Github Secrets, and reference them in Github Actions.
* Remove account ids from terraform var files.
* Restructure deployment to make use of secrets.

Closing #10 

## 💭 Motivation and context ##

Account ids are considered sensitive and should not be publicly accessible.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] _All_ future TODOs are captured in issues, which are referenced
      in code comments.
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [x] Tests have been added and/or modified to cover the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

File: .github/workflows/cool.yml [hunk 1]
```diff
@@ -1,182 +1,213 @@
...
+  # Github repositories
+  API_IMAGE_REPO: domain-manager-api
+  UI_IMAGE_REPO: domain-manager-ui
+
+  # Booleans
+  IS_STAGING: ${{ github.event_name == 'push' || github.event.action == 'cool_staging' || github.event.action == 'deploy' }}
+  IS_PRODUCTION: ${{ github.event.action == 'cool_production' }}
+  IS_PULL_REQUEST: ${{ github.event_name == 'pull_request' }}
+  BUILD_UI: github.event.client_payload.trigger == env.UI_IMAGE_REPO || github.event.action == 'cool_production'
+  BUILD_API: github.event.client_payload.trigger == env.API_IMAGE_REPO || github.event.action == 'cool_production'
```
Comment (dav3r) [inline .github/workflows/cool.yml]: I recommend alphabetizing this section.

File: .github/workflows/cool.yml [hunk 2]
```diff
@@ -1,182 +1,213 @@
...
     runs-on: ubuntu-latest
-    outputs:
-      ENVIRONMENT: ${{ steps.env.outputs.ENVIRONMENT }}
-      ASSUME_ROLE: ${{ steps.env.outputs.ASSUME_ROLE }}
-      ECR_URI: ${{ steps.env.outputs.ECR_URI }}
     steps:
+      # Install Terraform
+      - uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: 0.12.30
```
Comment (dav3r) [inline .github/workflows/cool.yml]: Might as well bump up to 0.12.31...
```suggestion
          terraform_version: 0.12.31
```

File: .github/workflows/cool.yml [hunk 3]
```diff
@@ -1,182 +1,213 @@
...
-
+        working-directory: ./cicd/cool
+        run: |
+          terraform plan \
+            -input=false \
+            -var-file ./vars/$ENVIRONMENT.tfvars \
+            -var="api_image_tag=$API_TAG" \
+            -var="ui_image_tag=$UI_TAG" \
+            -var="account_id=$ACCOUNT_ID" \
+            -var="shared_services_account_id=$SHARED_SERVICES_ACCOUNT_ID"
```
Comment (dav3r) [inline .github/workflows/cool.yml:199]: What is the reason for running `terraform plan` here?

File: cool/vars/production.tfvars [hunk 4]
```diff
@@ -1,10 +1,7 @@
 #=================================================
 #  PROVIDER
 #=================================================
-github_actions_role = "arn:aws:iam::779871898657:role/ProvisionAccount"
-shared_services_role = "arn:aws:iam::121087745329:role/ProvisionPrivateDNSRecords"
 workspace_type = "production"
-account_id = "779871898657"
```
Comment (dav3r) [inline cool/vars/production.tfvars:7]: Can the account ID also be removed from the Docker `image_url` below and in `staging.tfvars`?

**Review by dav3r — COMMENTED** [review]
This mostly looks great, but I have a few comments/suggestions/questions before I approve.

File: .github/workflows/cool.yml (hunk 3 above)
Comment (zenine07) [reply .github/workflows/cool.yml:199]: @dav3r The purpose of running terraform plan there is mostly for the pull requests. It is also helpful for when it runs `terraform apply` automatically in the pipeline as `terraform plan` gives more details as far as specifics of what changes. Look at the [latest pipeline run as an example](https://github.com/cisagov/domain-manager-cicd/runs/2632191008?check_suite_focus=true).

File: cool/vars/production.tfvars (hunk 4 above)
Comment (zenine07) [reply cool/vars/production.tfvars:7]: @dav3r Great catch. I totally missed that one. I will get that updated.

File: .github/workflows/cool.yml (hunk 1 above)
Comment (zenine07) [reply .github/workflows/cool.yml]: @dav3r That has been fixed in https://github.com/cisagov/domain-manager-cicd/pull/11/commits/51f7ebc7b70f2f823fc730df121e448f7ed0fa41

File: .github/workflows/cool.yml (hunk 2 above)
Comment (zenine07) [reply .github/workflows/cool.yml]: @dav3r Fixed in https://github.com/cisagov/domain-manager-cicd/pull/11/commits/f837a71e217904a2726fec1ac3a4581a493180ac

File: .github/workflows/cool.yml (hunk 3 above)
Comment (dav3r) [reply .github/workflows/cool.yml:199]: Got it, thanks.  I never use the `-auto-approve` switch with `terraform apply`, which apparently does not display the plan before executing it.

File: cool/vars/production.tfvars (hunk 4 above)
Comment (zenine07) [reply cool/vars/production.tfvars:7]: @dav3r Fixed in https://github.com/cisagov/domain-manager-cicd/pull/11/commits/b6bed6d26d8035e2635f150c25bf451dd27d2276

**Comment by dav3r** [comment]
@zenine07 - you should also turn on the branch protection in this and any other repos of yours that don't have it.  [This page](https://github.com/cisagov/development-guide/blob/develop/project_setup/branch-protection.md#branch-protection) shows the settings we use.

**Review by dav3r — APPROVED** [review]
Thanks for taking care of this @zenine07 - great work! 🚀

**Comment by zenine07** [comment]
> @zenine07 - you should also turn on the branch protection in this and any other repos of yours that don't have it. [This page](https://github.com/cisagov/development-guide/blob/develop/project_setup/branch-protection.md#branch-protection) shows the settings we use.

Great. I'll go take care of that.

**Comment by dav3r** [comment]
> > @zenine07 - you should also turn on the branch protection in this and any other repos of yours that don't have it. [This page](https://github.com/cisagov/development-guide/blob/develop/project_setup/branch-protection.md#branch-protection) shows the settings we use.
> 
> Great. I'll go take care of that.

@zenine07 Pinging on this as a reminder.

**Comment by zenine07** [comment]
> > > @zenine07 - you should also turn on the branch protection in this and any other repos of yours that don't have it. [This page](https://github.com/cisagov/development-guide/blob/develop/project_setup/branch-protection.md#branch-protection) shows the settings we use.
> > 
> > 
> > Great. I'll go take care of that.
> 
> @zenine07 Pinging on this as a reminder.

@dav3r Thanks for the reminder. Completed now.

**Comment by dav3r** [comment]
> > > > @zenine07 - you should also turn on the branch protection in this and any other repos of yours that don't have it. [This page](https://github.com/cisagov/development-guide/blob/develop/project_setup/branch-protection.md#branch-protection) shows the settings we use.
> > > 
> > > 
> > > Great. I'll go take care of that.
> > 
> > 
> > @zenine07 Pinging on this as a reminder.
> 
> @dav3r Thanks for the reminder. Completed now.

Excellent, thanks! 👍

**Review by jsf9k — APPROVED** [review]
This looks good to me.  Thanks for doing all this.

**Pull request merged**